src/main.cpp

```diff
@@ -1082,69 +1068,3 @@ QString UsageArguments ( char** argv )
     ).arg( argv[0] );
     // clang-format on
 }
-
-bool GetFlagArgument ( char** argv, int& i, QString strShortOpt, QString strLongOpt )
-{
-    if ( ( !strShortOpt.compare ( argv[i] ) ) || ( !strLongOpt.compare ( argv[i] ) ) )
-    {
-        return true;
-    }
-    else
-    {
-        return false;
-    }
-}
-
-bool GetStringArgument ( int argc, char** argv, int& i, QString strShortOpt, QString strLongOpt, QString& strArg )
-{
-    if ( ( !strShortOpt.compare ( argv[i] ) ) || ( !strLongOpt.compare ( argv[i] ) ) )
-    {
-        if ( ++i >= argc )
-        {
-            qCritical() << qUtf8Printable ( QString ( "%1: '%2' needs a string argument." ).arg ( argv[0] ).arg ( argv[i - 1] ) );
-            exit ( 1 );
-        }
-
-        strArg = argv[i];
-
-        return true;
-    }
-    else
-    {
-        return false;
-    }
-}
-
-bool GetNumericArgument ( int     argc,
-                          char**  argv,
-                          int&    i,
-                          QString strShortOpt,
-                          QString strLongOpt,
-                          double  rRangeStart,
-                          double  rRangeStop,
-                          double& rValue )
-{
-    if ( ( !strShortOpt.compare ( argv[i] ) ) || ( !strLongOpt.compare ( argv[i] ) ) )
-    {
-        QString errmsg = "%1: '%2' needs a numeric argument from '%3' to '%4'.";
-        if ( ++i >= argc )
-        {
-            qCritical() << qUtf8Printable ( errmsg.arg ( argv[0] ).arg ( argv[i - 1] ).arg ( rRangeStart ).arg ( rRangeStop ) );
-            exit ( 1 );
-        }
-
-        char* p;
-        rValue = strtod ( argv[i], &p );
-        if ( *p || ( rValue < rRangeStart ) || ( rValue > rRangeStop ) )
-        {
-            qCritical() << qUtf8Printable ( errmsg.arg ( argv[0] ).arg ( argv[i - 1] ).arg ( rRangeStart ).arg ( rRangeStop ) );
-            exit ( 1 );
-        }
-
-        return true;
-    }
-    else
-    {
-        return false;
-    }
-}
```

Comment:
These functions are now implemented in CCommandline.

Note that in the new implementation the 
```
qCritical() << qUtf8Printable ( ... );
exit ( 1 );
``` 
parts are replaced by the CCommandline::onArgumentError function which can, optionally, be passed in the CCommandLine constructor, since we probably don't want the program to exit on parsing wrong (--special?) parameters.